src/transport/SecureMessageCodec.cpp

```diff
@@ -71,7 +71,7 @@ CHIP_ERROR Encode(NodeId localNodeId, Transport::PeerConnectionState * state, Pa
     uint16_t totalLen = msgBuf->TotalLength();
 
     MessageAuthenticationCode mac;
-    ReturnErrorOnFailure(state->GetSecureSession().Encrypt(data, totalLen, data, packetHeader, mac));
+    ReturnErrorOnFailure(state->GetSenderSecureSession().Encrypt(data, totalLen, data, packetHeader, mac));
```

Comment:
Maybe this is a transitory state, but what I was expecting is that there is only one secure session that has an encryption key and a decryption key and uses the appropriate one for encrypt/decrypt.

That is, deriving a secure session takes both `kSpake2pI2RSessionInfo` and `kSpake2pR2ISessionInfo` and uses them as needed to derive the two keys the session stores....

Reply:
The keys are derived/created only when the secure session object is created. Are you suggesting that we should move the key derivation for both keys into a single function?

Reply:
I understand the setup.  I'm asking whether that's the setup we want to have.  With that setup, by just writing the wrong thing in a consumer of the PeerConnectionState you can end up encrypting with the wrong key (the one for receiving), right?

Maybe the right answer is to make the secure session getters protected and expose Encrypt/Decrypt APIs on the PeerConnectionState, which then selects the right thing?

Reply:
Updated the PR to add encrypt/decrypt API in PeerConnectionState.